ci_tools/check_pylint_commands.py

```diff
@@ -61,47 +102,98 @@ def check_expected_pylint_disable(file, disabled, flag, messages):
     args = parser.parse_args()
 
     diff = get_diff_as_json(args.diffFile)
+    print(diff)
 
     folder = args.folder
 
     files = [os.path.join(root,f) for root, dirs, filenames in os.walk(folder) for f in filenames if os.path.splitext(f)[1] == '.py']
 
     success = True
 
-    messages = []
+    messages = {"title":"Pylint Interaction","summary":"","annotations":[]}
 
     for f in files:
+        print("-------------", f, "-------------")
         with open(f, encoding="utf-8") as myfile:
             lines = [l.replace(' ','') for l in myfile.readlines()]
-        pylint_lines = [l.strip() for l in lines if l.startswith('#pylint:disable=')]
+        pylint_lines_and_numbers = [(l.strip(), i) for i,l in enumerate(lines,1) if l.startswith('#pylint:disable=')]
         disabled = set()
-        for l in pylint_lines:
-            disabled.update(l.split('=')[1].split(','))
+        for value, key in pylint_lines_and_numbers:
+            disabled.update([(tuple(value.split('=')[1].split(',')), key)])
         for r,d in accepted_pylint_commands.items():
             if r.match(f):
                 for di in d:
                     disabled.discard(di)
         p = pathlib.Path(f)
         if p.parts[0] == 'tests':
-            check_expected_pylint_disable(f, disabled, 'missing-function-docstring', messages)
-            check_expected_pylint_disable(f, disabled, 'missing-module-docstring', messages)
-            check_expected_pylint_disable(f, disabled, 'missing-class-docstring', messages)
+            msg = []
+            check_expected_pylint_disable(f, disabled, 'missing-function-docstring', msg)
+            check_expected_pylint_disable(f, disabled, 'missing-module-docstring', msg)
+            check_expected_pylint_disable(f, disabled, 'missing-class-docstring', msg)
+            first_iteration = True
+            for item in msg:
+                if first_iteration:
+                    messages['summary'] += item['summary'] + ' ' + item['flags']
+                    first_iteration = False
+                else:
+                    messages['summary'] += ', ' + item['flags']
+                messages['annotations'].append(item['annotations'])
+            if not first_iteration:
+                messages['summary'] += f' in {f}\n\n'
             if p.parts[1] == 'epyccel':
                 disabled.discard('reimported')
         if disabled:
             file_changed = f in diff
-            disabled_str = ", ".join(f"`{d}`" for d in disabled)
+            print(disabled, file_changed)
+            first_iteration = True
             if file_changed:
```

Comment:
The problem is here. The paths in `diff` and in `files` are not formatted the same. So we have diff: `{"ci_tools/check_pylint_commands.py" : ..., "tests/epyccel/test_epyccel_mod.py" : ...}`
but the file is `"./tests/epyccel/test_epyccel_mod.py"` which is not in the keys. To get round this I think we should always save paths as `Path` objects (see https://docs.python.org/3/library/pathlib.html) that way we are sure we are comparing things correctly

Reply:
Probably this is also broken in the `pyccel/pyccel` repo, I don't think there has been an example since I implemented this test

Reply:
I wonder if path problems are also responsible for the annotations not showing up. The example in the docs doesn't have "./" at the beginning